pydantic_ai_slim/pydantic_ai/mcp.py

```diff
@@ -149,36 +152,87 @@ async def call_tool(
         else:
             return content[0] if len(content) == 1 else content
 
-    async def __aenter__(self) -> Self:
-        if self._running_count == 0:
-            self._exit_stack = AsyncExitStack()
-
-            self._read_stream, self._write_stream = await self._exit_stack.enter_async_context(self.client_streams())
-            client = ClientSession(
-                read_stream=self._read_stream,
-                write_stream=self._write_stream,
-                sampling_callback=self._sampling_callback if self.allow_sampling else None,
-                logging_callback=self.log_handler,
+    async def _call_tool(
```

Comment:
feels weird to have the "private" call_tool call the "public" one. I guess it's okay but I usually like to use different names in these cases. something like `_call_or_process_tool` might work here?